src/research/index.js

```diff
@@ -29,67 +29,65 @@ export const sources = [
   semantic,
   stardust,
   w3,
-]
+].map(source => ({
+  ...source,
+  components: source.components.map(component => ({
+    ...component,
+    source: source.name,
+    openuiName: component.openuiName || component.name,
+    concepts: _.map(component.concepts, concept => ({
+      ...concept,
+      source: source.name,
+      component: component.openuiName || component.name,
+      openuiName: concept.openuiName || concept.name,
```

Comment:
Consider making a fn body and computing source and open ui name at the top.  Ease of reading and updating.